spec/services/post_alerter_spec.rb

```diff
@@ -2004,4 +2004,33 @@ def create_post_with_incoming
       expect(liked_notification.data_hash[:custom_key]).to eq(custom_data)
     end
   end
+
+  it "does not create notifications for PMs if not invited" do
+    SiteSetting.pm_tags_allowed_for_groups = "#{Group::AUTO_GROUPS[:everyone]}"
+    tag_1 = Fabricate(:tag)
+    TagUser.change(admin.id, tag_1.id, TagUser.notification_levels[:watching_first_post])
+    tag_2 = Fabricate(:tag)
+    TagUser.change(admin.id, tag_2.id, TagUser.notification_levels[:watching])
+
+    post = create_post(tags: [tag_1.name, tag_2.name], archetype: Archetype.private_message, target_usernames: "#{evil_trout.username}")
+    expect { PostAlerter.new.after_save_post(post, true) }.to change { Notification.count }.by(1)
+
+    expect(
+      Notification.exists?(
+        user: evil_trout,
+        notification_type: Notification.types[:private_message],
+        topic: post.topic,
+        post_number: 1,
+      ),
+    ).to eq(true)
+
+    expect(
+      Notification.exists?(
+        user: admin,
+        notification_type: Notification.types[:watching_first_post],
+        topic: post.topic,
+        post_number: 1,
+      ),
+    ).to eq(false)
```

Comment:
Is this check necessary since we're already expecting only one new notification record?